dev-docs/howtoguides/how_to_setup_windows_machine_for_wsl_testing.md

```diff
@@ -0,0 +1,158 @@
+## Installing winget
+
+For each Ubuntu distro, we will get the installers directly from the Microsoft store.
+To install such installers, we need to configure *winget*.
+
+To install *winget* on the machine, follow this [tutorial](https://winget.pro/winget-install-powershell/)
+
+```{note}
+if you see the following error when installing *winget*, restart the machine first:
+
+Add-AppxPackage : Deployment failed with HRESULT: 0x80073D02, The package could not be installed
+because resources it   modifies are currently in use. 
+```
+
+## Installing wsl
+
+To install *wsl*, just run the following command, as an administrator, on powershell:
+
+```console
+$ wsl --install
+```
+
+After that, we need to update WSL to the pre-release version by running:
+
+```console
+$ wsl --update --pre-release
+```
+
+Now, install a Bionic instance to test the whole WSL setup is working as expected:
+
+```console
+$ winget install --id "9PNKSF5ZN4SW" --accept-source-agreements --accept-package-agreements --silent
+```
+
+Now, create the image with:
+
+```console
+$ & ubuntu1804.exe install --root --ui=none
+```
+
+After that, just test to see if we can run commands on the WSL instance:
+
+```console
+$ wsl -d Ubuntu-18.04 --exec lsb_release -a
+```
+
+## Installing OpenSSH server
+
+To install OpenSSH server, follow these steps:
+
+1) Open *Settings*
+2) Go to *Apps*
+3) Go to *Optional Features*
+4) Go to *Add an optional feature*
+5) Search for *ssh*
+6) Install OpenSSH Server
+
+After installing the Service, open PowerShell and start-it by running:
+
+```console
+$ Start-Service sshd
+```
+
+And set it to automatically start after boot:
+
+```console
+$ Set-Service -Name sshd -StartupType 'Automatic'
+```
+
+### Configure the SSH keys
+
+We need to add a SSH public key into the Windows machine.
+
+To do that, follow these steps:
+
+1) Send the public key to the Windows machine:
+
+```console
+$ scp PUB_KEY_PATH ubuntu@IP:%programdata%/ssh
+```
+
+2) SSH into the Windows machine
+3) Run the following commands:
+
+```console
+$ cd %programdata%/ssh
+```
+
+```console
+$ type PUB_KEY_NAME >> administrators_authorized_keys
+```
+
+```console
+$ icacls administrators_authorized_keys /inheritance:r /grant "Administrators:F" /grant "SYSTEM:F"
+```
+
+Now test SSH into the machine using your private key
+
+## Configure the machine to automatic logon
+
+To properly run *winget* commands after a machine reboot, an user
+needs to be already logged into the machine. To automate that process,
+follow this [tutorial](https://learn.microsoft.com/en-us/troubleshoot/windows-server/user-profiles-and-logon/turn-on-automatic-logon#use-registry-editor-to-turn-on-automatic-logon)
+
+Note that you also need to manually create the *DefaultUserName* and *DefaultPassword* files
+
+## Running the integration test
+
+After this setup, you should be able to use that machine to run the WSL tests automatically.
+Before running a WSL test, remember to set the following environment variables:
```

Comment:
Set the environment variables inside each WSL instance subject to the integration tests or in the Windows host environment as well?

I suspect it's only the first, but wouldn't it be better to clarify it?

Reply:
It is actually neither of the options, we need to set up that up on your local machine. Our integration tests will use them to start and communicate with the Windows host machine and launch the WSL tests.

But I will make this more explicit